lib/mspec/helpers/ruby_exe.rb

```diff
@@ -128,9 +135,17 @@ def ruby_exe(code = :not_given, opts = {})
     code = tmpfile
   end
 
+  exception = opts.fetch(:exception, true)
+
   begin
     platform_is_not :opal do
-      `#{ruby_cmd(code, opts)}`
+      output = `#{ruby_cmd(code, opts)}`
+
+      if !$?.success? && exception
+        raise "ruby_exe(#{code}, #{opts}) failed: #{$?.inspect}"
```

Comment:
I would show the created command here, because it's easier to read and we need it anyway.
```ruby
command = ruby_cmd(code, opts)
output = `#{command}`

      if !$?.success? && exception
        raise "ruby_exe(#{command.inspect}) failed: #{$?.inspect}"

Reply:
That has the advantage that if there is some issue behavior in `ruby_cmd` it should be obvious from the output, and make it easier to run the command on its own.

Reply:
Done